lib/improver/cli/pattern.py

```diff
@@ -0,0 +1,276 @@
+                arg['default'] = None
+
+        if arg['type'] == 'str':
+            arg.pop('type')
+        elif arg['type'][:14] == 'iris.cube.Cube':
+            if name == 'input_cube':
+                arg.pop('type')
+                name = name[:-5]
+                arg['metavar'] = name.upper() + '_FILE'
+                name = name + '_filepath'
+                arg['help'] = 'A path to a NetCDF file ' + arg['help'][6:]
+            else:
+                continue
+        elif 'float' in arg['type']:
+            arg['type'] = float
+            if 'default' in arg:
+                try:
+                    arg['default'] = float(arg['default'])
+                except ValueError:
+                    if arg['default'] in consts:
+                        arg['default'] = consts[arg['default']]
+        elif arg['type'] == 'bool':
+            arg.pop('type')
+            arg.pop('metavar')
+            arg['action'] = 'store_true'
+            arg['default'] = False
+        elif 'int' in arg['type']:
+            arg['type'] = int
+            if 'default' in arg and arg['default'] is not None:
+                arg['default'] = int(arg['default'])
+        elif arg['type'] == 'numpy.int32':
+            arg['type'] = np.int32
+
+        if name == '--radii_by_lead_time':
+            arg['metavar'] = ('RADII_BY_LEAD_TIME', 'LEAD_TIME_IN_HOURS')
+            arg['nargs'] = 2
+            arg.pop('type')
+
+        if name == '--sum_or_fraction':
+            arg.pop('metavar')
+        parser.add_argument(name, **arg)
+
+
+def call_all(parser, argv, process_function, save_name, cube_args=None,
+             cubelist_args=None, option_cube_args=None, json_args=None):
+    """A function to load cubes, run function and save the cubes.
+
+    It starts by copying the ArgParser dictionary and removing the 'profile'
+    and 'profile_file' keys.
+    It checks that the extra strings given exists in the argParser.
+    It the removes the save filepaths from the arguments dicts and stores them
+    for later.
+    Sends to cubes and json off to be loaded and updated the dictionary.
+    Runs with function with all the dicts values as positional args.
+
+    Args:
+        parser (improver.argparser.ArgParser):
+            An argParser with the given arguments.
+        argv:
+            The arguments passed in straight from the command line.
+        process_function:
+            The process function to be used.
+        save_name (list of str):
+            A list of the ArgParser names for save files.
+        cube_args (list of str, optional):
+            A list of the ArgParser names for positional filepath.
+        cubelist_args (list of str, optional):
+            A list of argParser names for cubelists.
+        option_cube_args (list of str, optional):
+            A list of the ArgParser name for keyword arguments.
+        json_args (list of str, optional):
+
+    Returns:
+        (None)
```

Comment:
Maybe return the result, for good measure, in case someone cares to interactively use this